infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/GeometryService.kt

```diff
@@ -383,23 +393,30 @@ class GeometryService @Autowired constructor(
             val distanceInElement = distanceInSegment + (segment.sourceStart ?: 0.0)
             val distanceInGeometryAlignment = distanceInElement + (source.element?.staStart?.toDouble() ?: 0.0)
             source.profile?.getHeightAt(distanceInGeometryAlignment)
-        }?.let { trackMeterHeights ->
-            AlignmentHeights(
-                alignmentStartM = alignment.start?.m ?: return null,
-                alignmentEndM = alignment.end?.m ?: return null,
-                kmHeights = trackMeterHeights,
-                linkingSummary = linkingSummary,
-            )
         }
     }
 
+    fun getPlanAlignmentStartAndEnd(
+        planId: IntId<GeometryPlan>,
+        planAlignmentId: IntId<GeometryAlignment>,
+    ): AlignmentStartAndEnd? {
+        val planVersion = geometryDao.fetchPlanVersion(planId)
```

Comment:
Mielipideasiaa, mutta välireturnit tekee ainakin itselleni koodista vaikeampaa hahmottaa. Toki näin pienessä ja siistissä funkkarissa tuo ei suuremmin haittaa. Silti pitäisi itse enemmän yhden returnin mallista, ehkä vielä ?.letin kera jotta ei tarvitse iffitellä ja säilyy tuo oikosulkeminen jos sillä on merkitystä:
```
  val planVersion = geometryDao.fetchPlanVersion(planId)
  val plan = geometryDao.fetchPlan(planVersion)
  return plan.trackNumberId
      ?.let{ tnId -> geocodingService.getGeocodingContext(tnId, planVersion) }
      ?.let{ context ->
        planLayoutCache.getPlanLayout(planVersion).first
           ?.alignments?.find { alignment -> alignment.id == planAlignmentId }
           ?.let(geocodingContext::getStartAndEnd)
     }
```